pom.xml

```diff
@@ -39,6 +39,11 @@
             <artifactId>github-branch-source</artifactId>
             <version>2.4.0</version>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.mylyn.github</groupId>
+            <artifactId>org.eclipse.egit.github.core</artifactId>
+            <version>5.2.0.201812061821-r</version>
+        </dependency>
```

Comment:
I would really like to avoid the extra github dependency if we can. I'm investigating how to get the collaborator names and how the multibranch plugin handles this for `readTrusted` (since it basically does the same checks) which boils down to the `getTrustedRevision` [method](https://github.com/jenkinsci/github-branch-source-plugin/blob/0c05abdf1212df7b0f47dfe8dcbd804b5a81f6c0/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L1887).

I believe you can use [this GHRepository call](https://github.com/jenkinsci/github-branch-source-plugin/blob/0c05abdf1212df7b0f47dfe8dcbd804b5a81f6c0/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L1612) to get the collaborator names. The process for getting a GHRepository is shown [here](https://github.com/jenkinsci/github-branch-source-plugin/blob/0c05abdf1212df7b0f47dfe8dcbd804b5a81f6c0/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java#L1997).

I realize this means a rewrite of the GitHub helper, but I think it will keep us closer to the github branch source plugin functionality and be more future proof.

Reply:
I don't think I have the time to take that chunk of work on :(

Reply:
I can help test if it's necessary though

Reply:
Ok, I'll see if I can get some time this week to take it on. It shouldn't be too difficult as you've done all the plumbing, it's just changing that one method.

Reply:
ok thanks! Let me know if you can't update the branch (you should be able to)

Reply:
If you need GitHub API this dependency would be a better choice 
https://plugins.jenkins.io/github-api/